HandmadeMath.h

```diff
@@ -1493,29 +1556,10 @@ static inline HMM_Mat4 HMM_AddM4(HMM_Mat4 Left, HMM_Mat4 Right)
 
     HMM_Mat4 Result;
 
-#ifdef HANDMADE_MATH__USE_SSE
-    Result.Columns[0].SSE = _mm_add_ps(Left.Columns[0].SSE, Right.Columns[0].SSE);
-    Result.Columns[1].SSE = _mm_add_ps(Left.Columns[1].SSE, Right.Columns[1].SSE);
-    Result.Columns[2].SSE = _mm_add_ps(Left.Columns[2].SSE, Right.Columns[2].SSE);
-    Result.Columns[3].SSE = _mm_add_ps(Left.Columns[3].SSE, Right.Columns[3].SSE);
-#else
-    Result.Elements[0][0] = Left.Elements[0][0] + Right.Elements[0][0];
-    Result.Elements[0][1] = Left.Elements[0][1] + Right.Elements[0][1];
-    Result.Elements[0][2] = Left.Elements[0][2] + Right.Elements[0][2];
-    Result.Elements[0][3] = Left.Elements[0][3] + Right.Elements[0][3];
-    Result.Elements[1][0] = Left.Elements[1][0] + Right.Elements[1][0];
-    Result.Elements[1][1] = Left.Elements[1][1] + Right.Elements[1][1];
-    Result.Elements[1][2] = Left.Elements[1][2] + Right.Elements[1][2];
-    Result.Elements[1][3] = Left.Elements[1][3] + Right.Elements[1][3];
-    Result.Elements[2][0] = Left.Elements[2][0] + Right.Elements[2][0];
-    Result.Elements[2][1] = Left.Elements[2][1] + Right.Elements[2][1];
-    Result.Elements[2][2] = Left.Elements[2][2] + Right.Elements[2][2];
-    Result.Elements[2][3] = Left.Elements[2][3] + Right.Elements[2][3];
-    Result.Elements[3][0] = Left.Elements[3][0] + Right.Elements[3][0];
-    Result.Elements[3][1] = Left.Elements[3][1] + Right.Elements[3][1];
-    Result.Elements[3][2] = Left.Elements[3][2] + Right.Elements[3][2];
-    Result.Elements[3][3] = Left.Elements[3][3] + Right.Elements[3][3];
-#endif
+    Result.Columns[0] = HMM_AddV4(Left.Columns[0], Right.Columns[0]);
+    Result.Columns[1] = HMM_AddV4(Left.Columns[1], Right.Columns[1]);
+    Result.Columns[2] = HMM_AddV4(Left.Columns[2], Right.Columns[2]);
+    Result.Columns[3] = HMM_AddV4(Left.Columns[3], Right.Columns[3]);
```

Comment:
While I would normally appreciate the simplification, the purpose of the verbose operations here was to make as much information visible to compilers as possible. The more we nest functions for our own convenience, the more we pessimize debug builds and the less likely compilers optimize nicely.

This goes for all these simple operations that you modified to work in terms of vectors.